packages/@aws-cdk/aws-iam/lib/identity-base.ts

```diff
@@ -1,18 +1,36 @@
 import { IResource } from '@aws-cdk/core';
 import { IManagedPolicy } from './managed-policy';
-import { Policy } from "./policy";
+import { IPolicy, PolicyProps } from './policy';
 import { IPrincipal } from "./principals";
 
 /**
  * A construct that represents an IAM principal, such as a user, group or role.
  */
 export interface IIdentity extends IPrincipal, IResource {
   /**
-   * Attaches an inline policy to this principal.
-   * This is the same as calling `policy.addToXxx(principal)`.
-   * @param policy The policy resource to attach to this principal [disable-awslint:ref-via-interface]
+   * Creates and attaches a Policy to this principal, if it is mutable.
+   * If this principal is immutable, does nothing, and returns undefined.
```

Comment:
This docstring can't be correct anymore.

Reply:
Yes. I'm actually struggling to write these docs, like I said here: https://github.com/aws/aws-cdk/pull/4501#discussion_r337834507